nncf/structures.py

```diff
@@ -117,3 +118,35 @@ def __init__(self, data_loader: DataLoader,
     @classmethod
     def get_id(cls) -> str:
         return "autoq_precision_init_args"
+
+
+class LeGRInitArgs(NNCFExtraConfigStruct):
+    def __init__(self,
+                 train_loader,
+                 train_fn,
+                 val_loader,
+                 val_fn,
+                 train_optimizer,
+                 nncf_config: 'NNCFConfig'):
+        self.train_loader = train_loader
+        self.train_steps_fn = train_fn
+        self.val_loader = val_loader
+        self.val_fn = val_fn
+        self.train_optimizer = train_optimizer
+        self.config = nncf_config
+
+    @classmethod
+    def get_id(cls) -> str:
+        return "legr_init_args"
+
+
+class DistributedCallbacksArgs(NNCFExtraConfigStruct):
+    def __init__(self,
+                 wrapping_callback: Callable,
+                 unwrapping_callback: Callable):
+        self.wrap_model = wrapping_callback
+        self.unwrap_model = unwrapping_callback
+
+    @classmethod
+    def get_id(cls) -> str:
+        return "distributed_callbacks_args"
```

Comment:
Both of these structs should be extensively documented via docstrings. Also, specify the input and output types for the `Callable`s.

Reply:
Done